gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureLocalGenerateTask.java

```diff
@@ -16,17 +16,41 @@
 
 package com.palantir.gradle.conjure;
 
+import com.palantir.sls.versions.OrderableSlsVersion;
+import groovy.lang.Tuple;
+import groovy.lang.Tuple2;
 import java.io.File;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.gradle.api.tasks.CacheableTask;
 
 @CacheableTask
 public abstract class ConjureLocalGenerateTask extends ConjureGeneratorTask {
 
+    protected static final Pattern PATTERN = Pattern.compile("^([^.]+)-(.+?)(\\.conjure)?\\.json$");
+
+    static Tuple2<String, OrderableSlsVersion> parseProductNameAndVersion(String filename) {
```

Comment:
Nit: Instead of returning a groovy tuple, should we just define a small named wrapper within this class? I.e. a package-private immutables one? That way typing is a bit more explicit.

Reply:
We're very close! But Gradle plugins still need to stay on j11 for a little longer 🤞

Reply:
ack! switched to immutables in https://github.com/palantir/gradle-conjure/pull/1416/commits/7ea5012e14974d4aeffb2aac1fb3e47b245b51b7